### PR TITLE
Implement syscall buffering for RDTSC

### PR DESCRIPTION
**Pull request by rocallahan**
Before:
[roc@localhost rr]$ time rr record ~/rr/obj/bin/rdtsc_loop
rr: Saving execution to trace directory `/home/roc/.local/share/rr/rdtsc_loop-19'.
EXIT-SUCCESS
real    8m15.591s
user    5m9.203s
sys     3m10.668s

After:
[roc@localhost rr]$ time rr record ~/rr/obj/bin/rdtsc_loop
rr: Saving execution to trace directory `/home/roc/.local/share/rr/rdtsc_loop-18'.
EXIT-SUCCESS
real    0m7.757s
user    0m12.862s
sys     0m3.769s

FYI, without rr:
[roc@localhost rr]$ time ~/rr/obj/bin/rdtsc_loop
EXIT-SUCCESS
real    0m0.063s
user    0m0.060s
sys     0m0.003s

### CODE REVIEW

**Comment by rocallahan** [comment]
This is a reasonable-sized change so @khuey may want to look at it...

**Comment by rocallahan** [comment]
For RDTSC syscallbuf should preserve all registers, including flags. It's not hard to do on x86 so I've added that as a separate commit.

File: src/Registers.h [hunk 1]
```diff
@@ -360,6 +360,11 @@ class Registers {
   bool set_cx(uintptr_t value) { return RR_SET_REG_X86(ecx, rcx, value); }
 
   uintptr_t ax() const { return RR_GET_REG_X86(eax, rax); }
+  bool set_ax(uintptr_t value) { return RR_SET_REG_X86(eax, rax, value); }
```
Comment (khuey) [inline src/Registers.h:363]: set_ax is not actually used afaict

**Review by khuey — APPROVED** [review]

**Comment by khuey** [comment]
@Keno this failure looks like a CI issue

**Comment by Keno** [comment]
@DilumAluthge? Perhaps the branch just needs a rebase on top of the recent CI changes?

**Comment by DilumAluthge** [comment]
@staticfloat broke his promise to me 😉 

```
JULIA_CPU_THREADS: parameter null or not set
```

**Comment by DilumAluthge** [comment]
@staticfloat I'm guessing that the reason that `JULIA_CPU_THREADS` isn't defined on this agent is because it's the exclusive machine, and thus jobs are allowed to use every core on the machine.

In that case, I'd actually like to prevent the rr pipeline from running jobs on the exclusive agent. We don't need an entire machine for this pipeline, so I'd prefer to let other jobs (that actually need the entire machine) use the exclusive agent.

**Comment by DilumAluthge** [comment]
Anyway, in the meantime, I'll retry the job and let's see if we can get scheduled on a different machine.

**Comment by DilumAluthge** [comment]
CI is green after a retry.

**Pull request merged**